src/colmap/scene/image.h

```diff
@@ -149,11 +163,29 @@ inline camera_t Image::CameraId() const { return camera_id_; }
 
 inline void Image::SetCameraId(const camera_t camera_id) {
   THROW_CHECK_NE(camera_id, kInvalidCameraId);
+  THROW_CHECK(!HasCameraPtr())
+      << "SetCameraId can only be used when no camera address is synced in the "
+         "image. Call UnsetCameraPtr() first";
   camera_id_ = camera_id;
 }
 
 inline bool Image::HasCamera() const { return camera_id_ != kInvalidCameraId; }
 
+inline struct Camera* Image::CameraPtr() const {
+  THROW_CHECK(HasCameraPtr());
+  return camera_;
+}
+
+inline void Image::SetCameraPtr(struct Camera* camera) {
+  THROW_CHECK_NE(camera->camera_id, kInvalidCameraId);
+  camera_ = camera;
+  camera_id_ = camera->camera_id;
```

Comment:
THROW_CHECK_EQ(camera_id_, camera->camera_id) ?

Reply:
Similarly, in the SetCameraId() method, we probably want to do `if (HasCameraPtr()) { THROW_CHECK_EQ(camera_id_, camera->camera_id)` ?

Reply:
There might be some confusion here.

The idea of the design is that ``camera_`` always has higher priority than ``camera_id_``, so the ``camera_id`` should change accordingly to the corresponding camera. Thus, ``SetCameraId()`` is disabled when ``camera_`` is present (see here): https://github.com/colmap/colmap/pull/2715/files#diff-665386c3e6e1ebf0f2b3f383a2925c7a970b2e256f2f889f27fb090065158fbdR166